src/data/finset/locally_finite.lean

```diff
@@ -54,31 +54,33 @@ alias Ioc_eq_empty_iff ↔ _ finset.Ioc_eq_empty
 @[simp] lemma Ioo_eq_empty (h : ¬a < b) : Ioo a b = ∅ :=
 eq_empty_iff_forall_not_mem.2 $ λ x hx, h ((mem_Ioo.1 hx).1.trans (mem_Ioo.1 hx).2)
 
-@[simp] lemma Icc_eq_empty_of_lt (h : b < a) : Icc a b = ∅ :=
-Icc_eq_empty h.not_le
+@[simp] lemma Icc_eq_empty_of_lt (h : b < a) : Icc a b = ∅ := Icc_eq_empty h.not_le
 
-@[simp] lemma Ico_eq_empty_of_le (h : b ≤ a) : Ico a b = ∅ :=
-Ico_eq_empty h.not_lt
+@[simp] lemma Ico_eq_empty_of_le (h : b ≤ a) : Ico a b = ∅ := Ico_eq_empty h.not_lt
 
-@[simp] lemma Ioc_eq_empty_of_le (h : b ≤ a) : Ioc a b = ∅ :=
-Ioc_eq_empty h.not_lt
+@[simp] lemma Ioc_eq_empty_of_le (h : b ≤ a) : Ioc a b = ∅ := Ioc_eq_empty h.not_lt
 
-@[simp] lemma Ioo_eq_empty_of_le (h : b ≤ a) : Ioo a b = ∅ :=
-Ioo_eq_empty h.not_lt
+@[simp] lemma Ioo_eq_empty_of_le (h : b ≤ a) : Ioo a b = ∅ := Ioo_eq_empty h.not_lt
 
 variables (a)
 
-@[simp] lemma Ico_self : Ico a a = ∅ :=
-by rw [←coe_eq_empty, coe_Ico, set.Ico_self]
+@[simp] lemma Ico_self : Ico a a = ∅ := by rw [←coe_eq_empty, coe_Ico, set.Ico_self]
 
-@[simp] lemma Ioc_self : Ioc a a = ∅ :=
-by rw [←coe_eq_empty, coe_Ioc, set.Ioc_self]
+@[simp] lemma Ioc_self : Ioc a a = ∅ := by rw [←coe_eq_empty, coe_Ioc, set.Ioc_self]
 
-@[simp] lemma Ioo_self : Ioo a a = ∅ :=
-by rw [←coe_eq_empty, coe_Ioo, set.Ioo_self]
+@[simp] lemma Ioo_self : Ioo a a = ∅ := by rw [←coe_eq_empty, coe_Ioo, set.Ioo_self]
 
-@[simp] lemma right_not_mem_Ico {a b : α} : b ∉ Ico a b :=
-by { rw [mem_Ico, not_and], exact λ _, lt_irrefl _ }
+variables {a b}
+
+lemma left_mem_Icc : a ∈ Icc a b ↔ a ≤ b := by simp only [mem_Icc, true_and, le_refl]
+lemma left_mem_Ico : a ∈ Ico a b ↔ a < b := by simp only [mem_Ico, true_and, le_refl]
+lemma right_mem_Icc : b ∈ Icc a b ↔ a ≤ b := by simp only [mem_Icc, and_true, le_refl]
+lemma right_mem_Ioc : b ∈ Ioc a b ↔ a < b := by simp only [mem_Ioc, and_true, le_refl]
```

Comment:
Should these be `simp`?

Reply:
They aren't in simp normal form. So we could but only if we tweak priorities.